plugins/ingest-useragent/src/main/java/org/elasticsearch/ingest/useragent/UserAgentProcessor.java

```diff
@@ -0,0 +1,249 @@
+                        uaDetails.put("os_major", uaClient.operatingSystem.v1);
+                    }
+                    break;
+                case OS_MINOR:
+                    if (uaClient.operatingSystem != null && uaClient.operatingSystem.v2 != null) {
+                        uaDetails.put("os_minor", uaClient.operatingSystem.v2);
+                    }
+                    break;
+                case DEVICE:
+                    if (uaClient.device != null && uaClient.device.name != null) {
+                        uaDetails.put("device", uaClient.device.name);
+                    }
+                    else {
+                        uaDetails.put("device", "Other");
+                    }
+                    break;
+            }
+        }
+
+        ingestDocument.setFieldValue(targetField, uaDetails);
+    }
+
+    /** To maintain compatibility with logstash-filter-useragent */
+    private String buildFullOSName(VersionedName operatingSystem) {
+        if (operatingSystem == null || operatingSystem.name == null) {
+            return null;
+        }
+
+        StringBuilder sb = new StringBuilder(operatingSystem.name);
+
+        if (operatingSystem.v1 != null) {
+            sb.append(" ");
+            sb.append(operatingSystem.v1);
+
+            if (operatingSystem.v2 != null) {
+                sb.append(".");
+                sb.append(operatingSystem.v2);
+
+                if (operatingSystem.v3 != null) {
+                    sb.append(".");
+                    sb.append(operatingSystem.v3);
+
+                    if (operatingSystem.v4 != null) {
+                        sb.append(".");
+                        sb.append(operatingSystem.v4);
+                    }
+                }
+            }
+        }
+
+        return sb.toString();
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+    
+    String getField() {
+        return field;
+    }
+
+    String getTargetField() {
+        return targetField;
+    }
+
+    Set<Property> getProperties() {
+        return properties;
+    }
+    
+    UserAgentParser getUaParser() {
+        return parser;
+    }
+
+    public static final class Factory extends AbstractProcessorFactory<UserAgentProcessor> {
+
+        private final Map<String, UserAgentParser> userAgentParsers;
+        
+        public Factory(Map<String, UserAgentParser> userAgentParsers) {
+            this.userAgentParsers = userAgentParsers;
+        }
+
+        @Override
+        public UserAgentProcessor doCreate(String processorTag, Map<String, Object> config) throws Exception {
+            String field = readStringProperty(TYPE, processorTag, config, "field");
+            String targetField = readStringProperty(TYPE, processorTag, config, "target_field", "useragent");
+            String regexFilename = readOptionalStringProperty(TYPE, processorTag, config, "regex_file");
```

Comment:
Maybe change this into: `String regexFilename = readStringProperty(TYPE, processorTag, config, "regex_file", IngestUserAgentPlugin.DEFAULT_PARSER_NAME);`

Then we don't need the `if (regexFilename != null) {` at line 206?